AirportTicketBookingExercise/Logic/Service/FlightService.cs

```diff
@@ -0,0 +1,164 @@
+            return strBuilder.ToString();
+        }
+
+        public bool ImportFlightData(string importPath)
+        {
+            List<Flight> importedFlightData = new List<Flight>();
+            using (TextFieldParser csvParser = new TextFieldParser(importPath))
+            {
+                csvParser.CommentTokens = new string[] { "#" };
+                csvParser.SetDelimiters(new string[] { "," });
+                csvParser.HasFieldsEnclosedInQuotes = true;
+                csvParser.ReadLine();
+                try
+                {
+                    while (!csvParser.EndOfData)
+                    {
+                        string[] fields = csvParser.ReadFields();
+                        Flight flight = AddFlight(fields);
+                        if (flight == null)
+                        {
+                            return false;
+                        }
+                        importedFlightData.Add(flight);
+                    }
+                    _flightRepo.AddFlights(importedFlightData);
+                    return true;
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"Error while importing flight data: {ex.ToString()}");
+                    return false;
+                }
+            }
+        }
+
+        private Flight? AddFlight(string[] fields)
+        {
+            try
+            {
+                var flight = new Flight
+                {
+                    FlightId = int.Parse(fields[0]),
+                    FlightName = fields[1],
+                    DepartureCountry = fields[2],
+                    DestinationCountry = fields[3],
+                    DepartureAirport = fields[4],
+                    ArrivalAirport = fields[5],
+                    DepartureDate = DateTime.ParseExact(
+                                        fields[6],
+                                        new[] { "MM/dd/yyyy", "M/d/yyyy", "M/dd/yyyy", "MM/d/yyyy" },
+                                        CultureInfo.InvariantCulture,
+                                        DateTimeStyles.None),
+                    EconomyPrice = Decimal.Parse(fields[7]),
+                    BuisnessPrice = Decimal.Parse(fields[8]),
+                    FirstClassPrice = Decimal.Parse(fields[9]),
+                    SeatCapacity = int.Parse(fields[10])
+                };
+                return flight;
+            }
+            catch
+            {
+                return null;
+            }
+
+        }
+
+        public List<Flight> Search(string[] filterInfo)
+        {
+            List<Flight> filteredFlights = new List<Flight>();
+            if (filterInfo.Length < 3)
+                return _flightRepo.GetFlights();
+
+            FilterParam SearchParam = filterInfo[1].ParseFilterParam();
+            string input = filterInfo[2];
+
+            var flights = _flightRepo.GetFlights();
+            filteredFlights = SearchParam switch
```

Comment:
we can filter by multiple fields, so can have nested where statements, so switch here won't be enough